getconf/base.py

```diff
@@ -23,7 +23,46 @@
     logger.addHandler(compat.NullHandler())
 
 
-ConfigKey = collections.namedtuple('ConfigKey', ['section', 'entry', 'envvar', 'doc'])
+_ConfigKey = collections.namedtuple(
+    'ConfigKey',
+    ['section', 'entry', 'envvar', 'doc', 'default', 'type_hint']
+)
+
+
+class ConfigKey(_ConfigKey):
+
+    def __hash__(self):
+        if self.type_hint == 'list' and self.default is not None:
+            # Make sure we can hash default
+            default = tuple(self.default)
+        else:
+            default = self.default
+        return hash((self.section, self.entry, self.envvar, self.doc, default, self.type_hint))
+
+    def as_ini_entry(self):
+        """Format as a commented entry in INI format (section is omitted)
+
+        ; type=str - The provided documentation
+        ;entry = default_value
+        """
+        # Remove any new line in self.doc
+        one_line_doc = ' '.join(self.doc.split())
```

Comment:
Well, there is also `\r` and maybe some indentation that would not make any sense anymore without newlines.
So `' '.join(self.doc.split())` seems like an easy solution.